### PR TITLE
Add excluded/included events to get_event_logs api

### LINKED ISSUE #37582: Allow better filtering of event logs (audit log)
### What do you see as an issue?

The REST API get event logs endpoint only allows filtering logs by a single event type. This only includes that one event type and all others are discarded. One cannot filter out even a single event type.

### Solving the problem

In order to use this fully, I think we need to be able to pass multiple event names and also specify if we are including or excluding them.

In the webserver we use `audit_view_excluded_events` and `audit_view_included_events`. We should have something similar for the rest API for a user to specify what they want to include and exclude.

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by bbovenzi**
Allow API users to specify what events to include or exclude when listing event logs

closes https://github.com/apache/airflow/issues/37582

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/api_connexion/endpoints/test_event_log_endpoint.py [hunk 1]
```diff
@@ -291,6 +291,38 @@ def test_should_filter_eventlogs_by_when(self, create_log_model, session):
...
+
+        session.add(log_model_3)
+        session.flush()
+        response = self.client.get(
+            "/api/v1/eventLogs?excluded_events=TEST_EVENT_1,TEST_EVENT_2",
+            environ_overrides={"REMOTE_USER": "test_granular"},
+        )
+        assert response.status_code == 200
+        response_data = response.json
+        assert len(response_data["event_logs"]) == 1
```
Comment (jedcunningham) [inline tests/api_connexion/endpoints/test_event_log_endpoint.py:315]:
```suggestion
        assert len(response_data["event_logs"]) == 1
        assert {"cli_scheduler"} == {x.event for x in response_data["event_logs"]}
```

(Untested)

File: tests/api_connexion/endpoints/test_event_log_endpoint.py [hunk 2]
```diff
@@ -291,6 +291,38 @@ def test_should_filter_eventlogs_by_when(self, create_log_model, session):
...
             assert {eventlog["event"] for eventlog in response.json["event_logs"]} == expected_eventlogs
 
+    def test_should_filter_eventlogs_by_included_events(self, create_log_model, session):
+        create_log_model(event="TEST_EVENT_1", when=self.default_time)
+        create_log_model(event="TEST_EVENT_2", when=self.default_time_2)
+        log_model_3 = Log(event="cli_scheduler", owner="root", extra='{"host_name": "e24b454f002a"}')
+        log_model_3.dttm = self.default_time_2
+
+        session.add(log_model_3)
+        session.flush()
```
Comment (jedcunningham) [inline tests/api_connexion/endpoints/test_event_log_endpoint.py]:
```suggestion
        for event in ["TEST_EVENT_1", "TEST_EVENT_2", "cli_scheduler"]:
            create_log_model(event=event, when=self.default_time)

```

Would this work instead?

File: tests/api_connexion/endpoints/test_event_log_endpoint.py [hunk 3]
```diff
@@ -291,6 +291,38 @@ def test_should_filter_eventlogs_by_when(self, create_log_model, session):
...
+        assert len(response_data["event_logs"]) == 2
+
+    def test_should_filter_eventlogs_by_excluded_events(self, create_log_model, session):
+        create_log_model(event="TEST_EVENT_1", when=self.default_time)
+        create_log_model(event="TEST_EVENT_2", when=self.default_time_2)
+        log_model_3 = Log(event="cli_scheduler", owner="root", extra='{"host_name": "e24b454f002a"}')
+        log_model_3.dttm = self.default_time_2
+
+        session.add(log_model_3)
+        session.flush()
```
Comment (jedcunningham) [inline tests/api_connexion/endpoints/test_event_log_endpoint.py]:
```suggestion
        for event in ["TEST_EVENT_1", "TEST_EVENT_2", "cli_scheduler"]:
            create_log_model(event=event, when=self.default_time)

```

File: tests/api_connexion/endpoints/test_event_log_endpoint.py [hunk 4]
```diff
@@ -291,6 +291,38 @@ def test_should_filter_eventlogs_by_when(self, create_log_model, session):
             assert response.status_code == 200
             assert {eventlog["event"] for eventlog in response.json["event_logs"]} == expected_eventlogs
 
+    def test_should_filter_eventlogs_by_included_events(self, create_log_model, session):
```
Comment (jedcunningham) [inline tests/api_connexion/endpoints/test_event_log_endpoint.py]:
```suggestion
    def test_should_filter_eventlogs_by_included_events(self, create_log_model):
```

If so, we can also get rid of the session fixture.

File: tests/api_connexion/endpoints/test_event_log_endpoint.py [hunk 5]
```diff
@@ -291,6 +291,38 @@ def test_should_filter_eventlogs_by_when(self, create_log_model, session):
...
+        session.flush()
+        response = self.client.get(
+            "/api/v1/eventLogs?included_events=TEST_EVENT_1,TEST_EVENT_2",
+            environ_overrides={"REMOTE_USER": "test_granular"},
+        )
+        assert response.status_code == 200
+        response_data = response.json
+        assert len(response_data["event_logs"]) == 2
+
+    def test_should_filter_eventlogs_by_excluded_events(self, create_log_model, session):
```
Comment (jedcunningham) [inline tests/api_connexion/endpoints/test_event_log_endpoint.py]:
```suggestion
    def test_should_filter_eventlogs_by_excluded_events(self, create_log_model):
```

File: tests/api_connexion/endpoints/test_event_log_endpoint.py [hunk 6]
```diff
@@ -291,6 +291,29 @@ def test_should_filter_eventlogs_by_when(self, create_log_model, session):
...
+    def test_should_filter_eventlogs_by_included_events(self, create_log_model):
+        for event in ["TEST_EVENT_1", "TEST_EVENT_2", "cli_scheduler"]:
+            create_log_model(event=event, when=self.default_time)
+        response = self.client.get(
+            "/api/v1/eventLogs?included_events=TEST_EVENT_1,TEST_EVENT_2",
+            environ_overrides={"REMOTE_USER": "test_granular"},
+        )
+        assert response.status_code == 200
+        response_data = response.json
+        assert len(response_data["event_logs"]) == 2
```
Comment (jedcunningham) [inline tests/api_connexion/endpoints/test_event_log_endpoint.py:303]:
```suggestion
        assert len(response_data["event_logs"]) == 2
        assert {"TEST_EVENT_1", "TEST_EVENT_2"} == {x["event"] for x in response_data["event_logs"]}
```

Oops, forgot this one in the last round.

**Review by ephraimbuddy — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Review by phanikumv — APPROVED** [review]

**Pull request merged**